ui/components/src/ui-components/viewport/ViewportComponent.tsx

```diff
@@ -163,13 +183,16 @@ export class ViewportComponent extends React.Component<ViewportProps, ViewportSt
       if (!viewState) {
         Logger.logError(UiComponents.loggerCategory(this), `View state failed to load`);
         return undefined;
+      } else {
+        return viewState;
       }
     } else {
       Logger.logError(UiComponents.loggerCategory(this), `Either viewDefinitionId or viewState must be provided as a ViewportComponent Prop`);
       return undefined;
     }
 
-    return viewState;
+    const clonedViewState = await ViewportComponent.cloneViewState(this.props.imodel, viewState);
+    return clonedViewState;
```

Comment:
nitpick - I believe you could have just done...
return ViewportComponent.cloneViewState(this.props.imodel, viewState);

Reply:
Correct, I could have. I added the extra line when I was debugging -- I'll change it.